### ISSUE TITLE
[Feature request]: Firmware OTA for Develco (Frient) HESZB-120 Intelligent Heat Alarm

### ISSUE THREAD

**Issue by Bloodpack** [Koenkk/zigbee2mqtt#21512]
### Is your feature request related to a problem? Please describe

First of all sorry to open an issue here, but i have no idea how to open a pull request on [zigbee-OTA repository](https://github.com/Koenkk/zigbee-OTA) 

***github noob here***

reference to https://github.com/Koenkk/zigbee2mqtt/issues/8718

I contacted Frient and asked if i can get the Firmware for  [Develco (Frient) HESZB-120 Intelligent Heat Alarm](https://www.zigbee2mqtt.io/devices/HESZB-120.html#develco-heszb-120)

they responded within 15min with the firmware

EDIT:
there are two files in the .zip `4.0.8` and `4.0.8_B` they said the one with the `B` is just mirrored

[HeatAlarm_4.0.8.zip](https://github.com/Koenkk/zigbee2mqtt/files/14356502/HeatAlarm_4.0.8.zip)

### Describe the solution you'd like

It would be nice if this firmware can get implemented in the OTA Firmware update

### Describe alternatives you've considered

.

### Additional context

I have asked if Frient could possibly contribute future firmware releases directly to the zigbee-OTA repository

Thanks and sorry again for not being able to open a pull request

**Comment by jamesonuk** [comment]
@Bloodpack There are two files in there (4.0.8 and 4.0.8_B)
Did they give you any indication of what the B file is for?   Is it a beta? (seems odd it would be called 4.0.8 if it was though)

**Comment by Bloodpack** [comment]
@jamesonuk 

They replied and said the following
```Hi again.

 

Just use “4.0.8”, the “4.0.8_B” is just a mirrored version of the other.

 

Best regards,

 ```

**Comment by jamesonuk** [comment]
I did a binary compare originally and it said the files were slightly different but having had a look, the firmware is already there https://github.com/Koenkk/zigbee-OTA/blob/master/images/Develco/FireAlarm_4.0.8.zigbee but the device is not configured for OTA.   I will create a PR in a minute

**Comment by Bloodpack** [comment]
@jamesonuk the Firmware that is located on here [](https://github.com/Koenkk/zigbee-OTA/blob/master/images/Develco/FireAlarm_4.0.8.zigbee ) is the Firmware for the SmokeAlarm Develco (Frient) SMSZB-120 and is called SmokeAlarm

the firmware that i have obtained today is for the Heat Alarm Develco (Frient) HESZB-120 and is called HeatAlarm

**Comment by jamesonuk** [comment]
> @jamesonuk the Firmware that is located on here https://github.com/Koenkk/zigbee-OTA/blob/master/images/Develco/FireAlarm_4.0.8.zigbee is the Firmware for the SmokeAlarm Develco (Frient) SMSZB-120 and is called FireAlarm
> 
> the firmware that i have obtained today is for the Heat Alarm Develco (Frient) HESZB-120 and is called HeatAlarm
> 
> The SMSZB-120 Smoke Detector and the HESZB-120 Heat detector are very similar, but i don´t know if the firmware is identical or not....

@Bloodpack The files inside the zip are called FireAlarm and I have confirmed it is identical to the one in the repo.
I have applied the PR above locally and I am currently updating one of my heat alarms
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1040621/14842135-3115-460b-b29b-b6e9e9d311d9)

**Comment by mamrai1** [comment]
I have updated one of my friend smart heat sensors to 4.0.8 from 3.425 and it seems like something is wrong with the z2m devixe converter or the new firmware.
When pressing alarm button - off on z2m device's webgui it sounds the alarm..

**Comment by Bloodpack** [comment]
> I have updated one of my friend smart heat sensors to 4.0.8 from 3.425 and it seems like something is wrong with the z2m devixe converter or the new firmware. When pressing alarm button - off on z2m device's webgui it sounds the alarm..

@jamesonuk can you have a look please if you have the same issue..... thank you

**Comment by jamesonuk** [comment]
> I have updated one of my friend smart heat sensors to 4.0.8 from 3.425 and it seems like something is wrong with the z2m devixe converter or the new firmware. When pressing alarm button - off on z2m device's webgui it sounds the alarm..

I think this is the correct behaviour (kind of).
This is not alarm off but the trigger for manual start of alarm.   I think you can't stop the alarm remotely by design.   I thought it worked that way in the previous firmware too.

I can't recall what it did before and I have never tried it through the official hub/app so not sure what it is meant to do (and the manual only says where to place it and doesn't even cover how to pair it with anything)

In fact searching came across https://github.com/Koenkk/zigbee2mqtt/issues/8321 which I apparently commented on :)   looks like this was HA (and smoke alarm) but it looks like it has always done this?

**Comment by Migaddo** [comment]
With the new firmware, i have pairing issues. It succusfully pairs, leaving the network and pair again in a loop.

**Comment by Rickth64** [comment]
> I have updated one of my friend smart heat sensors to 4.0.8 from 3.425 and it seems like something is wrong with the z2m devixe converter or the new firmware. When pressing alarm button - off on z2m device's webgui it sounds the alarm..

I have the exact same thing. With the previous firmware I could send off and on (or start or test or whatever you like to call it) and the alarm would adhere to that command. Now, sending 'off' triggers the alarm :(

**Comment by Rickth64** [comment]
@jamesonuk @mamrai1 It’s not expected behavior. Already fixed with Koenkk/zigbee-herdsman-converters#7179

**Comment by rafpigna** [comment]
Hello, sorry to reply on this issue, but I'm trying the OTA update, but It fails everytime.

It's also extremely slow, the ETA is about 4 hours to complete! But after about 20 minutes at about 10% it fails.
